protocol/nats_jetstream/v2/message.go

```diff
@@ -99,7 +99,24 @@ func (m *Message) ReadBinary(ctx context.Context, encoder binding.BinaryWriter)
 
 // Finish *must* be called when message from a Receiver can be forgotten by the receiver.
 func (m *Message) Finish(err error) error {
-	return nil
+	// Ignore error if message has already been Ack/Nak.
+	// Ack and Nak first checks to see if the message has been acknowleged
+	// and immediately returns an error without applying any logic to the message on the server.
+	// A message will auto acknowledge unless manual ack is set or policy is set to AckNonePolicy
+	// see: ManualAck() subscription option
+	// see: ConsumerConfig.AckPolicy
+	if err != nil {
+		errNak := m.Msg.Nak()
```

Comment:
What are other errors that `Ack` and `Nak` can return? Can you please run a local dummy test with a real NATS backend and always return an error here so we understand how the system behaves end-to-end? IMHO it should be fine (warning logged), but want to make sure this doesn't lead to an endless invocation loop since with this change we're changing the behavior for users.

Reply:
@stephen-totty-hpe ping

Reply:
I will close this PR if https://github.com/cloudevents/sdk-go/pull/1095 gets merged.